prog.go

```diff
@@ -338,19 +339,39 @@ func (p *Program) Clone() (*Program, error) {
 		return nil, fmt.Errorf("can't clone program: %w", err)
 	}
 
-	return &Program{p.VerifierLog, dup, p.name, p.typ}, nil
+	return &Program{p.VerifierLog, dup, p.name, "", p.typ}, nil
 }
 
 // Pin persists the Program past the lifetime of the process that created it
 //
 // This requires bpffs to be mounted above fileName. See https://docs.cilium.io/en/k8s-doc/admin/#admin-mount-bpffs
 func (p *Program) Pin(fileName string) error {
-	if err := internal.BPFObjPin(fileName, p.fd); err != nil {
-		return fmt.Errorf("can't pin program: %w", err)
+	if err := pin(p.pinnedPath, fileName, p.fd); err != nil {
+		return err
+	}
+	p.pinnedPath = fileName
+	return nil
+}
+
+// Unpin removes the persisted state for the Program.
```

Comment:
Nit: perhaps we could explain the user here that the kernel will remove the program when the host process exits, and that the unpin operation is executed on the path specified in the latest call to `Pin()`. 'persistent state for the Program' sounds a bit vague to me.

Reply:
It's really easy to go into the weeds here, I prefer a high level explanation personally. How about `Unpin reverses the effect of Pin.`? Maybe too much meta.

Reply:
Agreed, `Pin()` can be expanded a bit, so `Unpin()` can be kept short.

Reply:
>perhaps we could explain the user here that the kernel will remove the program when the host process exits, and that the unpin operation is executed on the path specified in the latest call to Pin()

This is not really a behavior defined by the library. I've kept the description as-is.

Reply:
IMO the library aids in educating the user about consequences and common pitfalls. My point is, 'removes the persisted state' is ambiguous and vague. There is no persisted state, as that would imply writing to disk.

Reply:
>There is no persisted state, as that would imply writing to disk.

Umm... Why is that? Even though BPF `pinning` involves `FD` ref counting behind the scenes in the Linux kernel, there is a persistent state on the BPF virtual file system from user space perspective. 

Do you think it'll be helpful if we call out that the persistent state is on the BPF virtual file system?

Reply:
I think it's good like it is now :+1: